elasticsearch_collectd.py

```diff
@@ -43,8 +43,261 @@
 
 COLLECTION_INTERVAL = 10
 
+INDEX_INTERVAL = 300
+
+INDEX_SKIP = 0
+
+SKIP_COUNT = 0
+
 CLUSTER_STATUS = {'green': 0, 'yellow': 1, 'red': 2}
 
+DETAILED_METRICS = True
+THREAD_POOLS = []
+CONFIGURED_THREAD_POOLS = set()
+
+
+DEFAULTS = {
+    # PAGE: Elasticsearch
+
+    # DASHBOARD: Elasticsearch Node
+
+    # CHART: Requests/sec
+    "indices.search.query-total",
+    "indices.indexing.index-total",
+    "indices.get.total",
+
+    # CHART: Document Growth Rate %
+    "indices.docs.count",
+    "indices.docs.count",
+    "indices.docs.count",
+    "indices.docs.count",
+
+    # CHART: Indexing Requests/sec
+    "indices.indexing.index-total",
+
+    # CHART: Average Query Latency (ms)
+    "indices.search.query-total",
+    "indices.search.query-time",
+
+    # CHART: File Descriptors and Segments
+    "process.open_file_descriptors",
+    "indices.segments.count",
+
+    # CHART: Memory Heap Size (bytes)
+    "jvm.mem.heap-used",
+    "jvm.mem.heap-committed",
+
+    # CHART: Active Merges
+    "indices.merges.current",
+
+    # CHART: Search Requests/sec
+    "indices.search.query-total",
+
+    # CHART: Merges/sec
+    "indices.merges.total",
+
+    # CHART: Thread Pool Rejections
+    "thread_pool.bulk.rejected",
+    "thread_pool.flush.rejected",
+    "thread_pool.generic.rejected",
+    "thread_pool.get.rejected",
+    "thread_pool.index.rejected",
+    "thread_pool.merge.rejected",
+    "thread_pool.optimize.rejected",
+    "thread_pool.refresh.rejected",
+    "thread_pool.search.rejected",
+    "thread_pool.snapshot.rejected",
+
+    # CHART: Cache Sizes (bytes)
+    "indices.cache.filter.size",
+    "indices.cache.field.size",
+
+    # CHART: GC Time %
+    "jvm.gc.time",
+    "jvm.uptime",
+
+    # DASHBOARD: Elasticsearch
+
+    # CHART: Top Clusters by Query Latency (ms)
+    "indices.search.query-total",
+    "indices.search.query-time",
+    "indices.search.query-total",
+    "indices.search.query-time",
+
+    # CHART: Top Clusters by Indexing Requests
+    "indices.indexing.index-total",
+
+    # CHART: Merges/sec
+    "indices.merges.total",
+
+    # CHART: # Hosts/Clusters
+    "indices.indexing.index-total",
+    "indices.indexing.index-total",
+
+    # CHART: Thread Pool Rejections
+    "thread_pool.bulk.rejected",
+    "thread_pool.flush.rejected",
+    "thread_pool.generic.rejected",
+    "thread_pool.get.rejected",
+    "thread_pool.index.rejected",
+    "thread_pool.merge.rejected",
+    "thread_pool.optimize.rejected",
+    "thread_pool.refresh.rejected",
+    "thread_pool.search.rejected",
+    "thread_pool.snapshot.rejected",
+
+    # CHART: Search Requests
+    "indices.search.query-total",
+
+    # CHART: GC Time %
+    "jvm.gc.time",
+    "jvm.uptime",
+    "jvm.gc.time",
```

Comment:
This was more for human readability and easy confirmation that the default metrics directly correspond to a default dashboard chart.  The duplicates were printed automatically because the metric is included in the chart multiple times.  Because it is loaded as a python set at run time the duplicates will simply be ignored.